modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessor.java

```diff
@@ -976,11 +977,31 @@ private void addConnectionInterceptors(InterceptorChain.Builder chainBuilder) {
   private void addCursorResetInterceptor(InterceptorChain.Builder chainBuilder) {
     List<String> streamParams = new ArrayList<>(5);
     componentModel.getAllParameterModels().forEach(
-                                                   p -> getType(p.getType(), getClassLoader(extensionModel))
-                                                       .filter(clazz -> InputStream.class.isAssignableFrom(clazz)
-                                                           || Iterator.class.isAssignableFrom(clazz))
-                                                       .ifPresent(clazz -> streamParams.add(p.getName())));
-
+                                                   p -> {
+                                                     getType(p.getType(), getClassLoader(extensionModel))
+                                                         .filter(clazz -> InputStream.class.isAssignableFrom(clazz)
+                                                             || Iterator.class.isAssignableFrom(clazz))
```

Comment:
What case is the "Iterator" covering? Can an operation receive an Iterator as a parameter directly? Could not find tests for this case. What would be an example so I can add one?